constraint/pkg/client/drivers/local/driver.go

```diff
@@ -71,14 +73,28 @@ type Driver struct {
 
 	// clientCertWatcher is a watcher for the TLS certificate used to communicate with providers.
 	clientCertWatcher *certwatcher.CertWatcher
+
+	// gatherMetrics controls whether the driver attempts to gather any metrics around its API calls
+	gatherMetrics bool
+
+	// statsEntries returns the current statistics observed since the last API call. At present
+	// statsEntries are only gathered on the Query() API.
+	statsEntries []*instrumentation.StatsEntry
 }
 
-// RegoEvaluationMeta has rego specific metadata from evaluation.
-type RegoEvaluationMeta struct {
-	// TemplateRunTime is the number of milliseconds it took to evaluate all constraints for a template.
-	TemplateRunTime float64 `json:"templateRunTime"`
-	// ConstraintCount indicates how many constraints were evaluated for an underlying rego engine eval call.
-	ConstraintCount uint `json:"constraintCount"`
+// StatsEntries returns any stats associated with the driver API if GatherMetrics
+// driver arg was passed in. The only current endpoinut supported is Query().
+func (d *Driver) StatsEntries() []*instrumentation.StatsEntry {
```

Comment:
If we must return the stats as the same time as the current `Query()` calls then this could break the backwards compatibility of the interface since we'd have to modify the return types.

We could add a `QueryWStats()` (or some better named func) as previously discussed that returns stats (in addition to what `Query` returns too).

I'm partial to the `QueryWStats()` avenue since we wouldn't necessarily have to modify the `Driver` interface to accommodate that func on the local driver and keep casting the driver in the client. As w iterate on Stats & the Driver this could help us shield the interface form multiple changes. Wdyt?

Reply:
No need for an extra function... this is `v0.x.x` for a reason.

Also, I don't think I agree with you that backwards compatibility necessarily needs to be broken, the mere altering of a struct doesn't break compatibility, it depends on how it's altered. See K8s backwards compatibility docs for some examples:

https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api_changes.md#on-compatibility

However, we __know__ we are breaking backwards compatibility in the sense that we are removing the previous stats work.

Reply:
Per this comment and offline chats, we decided to modify the `Query` func on the driver interface to return a `QueryResponse`. We'd embed the `StatsEntries` in that response.